### PR TITLE
[SHLWAPI][SDK] Implement SHPropertyBag_WriteDWORD etc.

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Implementing missing features...
JIRA issue: [CORE-9283](https://jira.reactos.org/browse/CORE-9283)

## Proposed changes

- Implement `SHPropertyBag_WriteBOOL`, `SHPropertyBag_WriteSHORT`, `SHPropertyBag_WriteLONG`, `SHPropertyBag_WriteDWORD`, `SHPropertyBag_WriteStr`, `SHPropertyBag_WriteGUID`, and `SHPropertyBag_WriteStream` functions.
- Modify `dll/win32/shlwapi/shlwapi.spec`.
- Modify `sdk/include/reactos/shlwapi_undoc.h`.

## TODO

- [x] Do big tests.

### CODE REVIEW

File: dll/win32/shlwapi/ordinal.c
Comment (HBelusca) [inline dll/win32/shlwapi/ordinal.c:5412]: Huh.... why `LPCOLESTR` instead of `LPCWSTR` ?

Comment (katahiromz) [reply dll/win32/shlwapi/ordinal.c:5412]: In some old systems, `OLECHAR` type is `CHAR`.

Comment (HBelusca) [reply dll/win32/shlwapi/ordinal.c:5412]: yes, but the same as TCHAR...

Those APIs use unicode strings, so they should keep the LPCWSTR instead of changing the parameter.

Comment (katahiromz) [reply dll/win32/shlwapi/ordinal.c]: Fixed.

File: sdk/include/reactos/shlwapi_undoc.h
Comment (JoachimHenze) [inline sdk/include/reactos/shlwapi_undoc.h:109]: LPCOLESTR to LPCWSTR also here?

File: dll/win32/shlwapi/ordinal.c
Comment (JoachimHenze) [inline dll/win32/shlwapi/ordinal.c:5441]: OLECHAR intended here? @HBelusca ?

File: sdk/include/reactos/shlwapi_undoc.h
Comment (katahiromz) [reply sdk/include/reactos/shlwapi_undoc.h:109]: Fixed.

File: dll/win32/shlwapi/ordinal.c
Comment (katahiromz) [reply dll/win32/shlwapi/ordinal.c]: Fixed as `WCHAR`.

**Review by JoachimHenze — APPROVED** [review]
Approved from my side. Please wait for @hbelusca s approval before merging.

**Comment by katahiromz** [comment]
KVM BEFORE: 00420af https://reactos.org/testman/compare.php?ids=88432
KVM AFTER: b4d5a3d https://reactos.org/testman/compare.php?ids=88450
https://reactos.org/testman/compare.php?ids=88432,88450
LGTM.

VBox BEFORE: 00420af https://reactos.org/testman/compare.php?ids=88412
VBox AFTER: b4d5a3d https://reactos.org/testman/compare.php?ids=88438
https://reactos.org/testman/compare.php?ids=88412,88438
`winhttp:winhttp` `CRASH`'ed. But this is unrelated. LGTM.

**Review by HBelusca — APPROVED** [review]

**Pull request merged**
